initialization/initialization.go

```diff
@@ -91,22 +91,18 @@ func (init *Initializer) Initialize() error {
 
 	init.stopChan = make(chan struct{})
 	init.doneChan = make(chan struct{})
+	init.numLabelsWrittenChan = make(chan uint64)
 
 	init.initializing = true
-
 	init.mtx.Unlock()
 
 	defer func() {
 		init.mtx.Lock()
 		init.initializing = false
-		init.mtx.Unlock()
 
 		close(init.doneChan)
-
-		if init.numLabelsWrittenChan != nil {
-			close(init.numLabelsWrittenChan)
-			init.numLabelsWrittenChan = nil
-		}
+		close(init.numLabelsWrittenChan)
+		init.mtx.Unlock()
```

Comment:
NIT, you could use `defer`.

Reply:
`defer` in a `defer func() {}`?

![image](https://i.imgflip.com/6yhz1i.jpg)